main.c

```diff
@@ -132,7 +132,29 @@ static void reset_devices(void) {
     #endif
 }
 
-STATIC void start_mp(supervisor_allocation *heap) {
+#if MICROPY_ENABLE_PYSTACK
+STATIC supervisor_allocation __attribute__ ((noinline)) * alloc_pystack(void) {
+    mp_int_t pystack_size = CIRCUITPY_PYSTACK_SIZE;
+    #if CIRCUITPY_OS_GETENV
+    // Fetch value if exists from settings.toml
+    // Leaves size to build default on any failure
+    (void)common_hal_os_getenv_int("CIRCUITPY_PYSTACK_SIZE", &pystack_size);
+    // Check if value is valid
+    if ((CIRCUITPY_PYSTACK_SIZE != pystack_size) && ((pystack_size < 384) || (pystack_size % sizeof(size_t) != 0))) {
```

Comment:
I would write this as:
```
if (getenv_err != GETENV_OK) {
    serial_write_compressed(translate("\nWARNING: Invalid CIRCUITPY_PYSTACK_SIZE, defaulting back to build value.\n\n"));
        pystack_size = CIRCUITPY_PYSTACK_SIZE;
}
pystack_size = MAX(pystack_size, CIRCUITPY_MIN_PYSTACK_SIZE);
```
Because:
1. You need to check for a parsing error in `common_hal_os_getenv_int`.
2. I don't think `pystack` not being mod `sizeof(size_t)` is a real error. Just let integer division round it down.
3. I assume that 384 is the minimum stack size, which should be put in a #define CIRCUITPY_MIN_PYSTACK_SIZE. And again, I don't think being outside this is a error of note.

Reply:
1. No, if for any reason it fails, it will not update the value. The same thing was done on web workflow (from which I copypasta'ed).
2. Uhh, it sure could be rounded down, as a matter of fact imma do it right now.
3. Yea, a guestimation on my part. Wouldn't call it the ideal value but fine. Someone should test that.

Reply:
Re #1: but then the user doesn't get an error message about their settings file being bad

Reply:
Invalid settings.toml is something ignored.
All current getenv code ignores errors (web-workflow & bt-workflow).

Reply:
Despite precedent, I don't think that is a good practice. If I make a typo in my settings, I'd benefit from a diagnostic telling me so, instead of scratching my head wondering why nothing happened.

Reply:
I think just  "Invalid CIRCUITPY_PYSTACK_SIZE" is probably good enough, and it uses existing strings; I agree ignoring is not so great. It will push the user to documentation.

Reply:
`CIRCUITPY_PYSTACK_SIZE` has never been used as a string before. It's just a nameshake with the `#if`.
The plan was to get rid of all the compile-time settings for stacks in favor of this implementation.
However they both remained in the end.
I do not see a better name that isn't also used by `#if`'s.
And there is no real risk of users confusing compilation settings with settings.toml variables.
Pretty much nobody will be using stackless after this pr is merged.
There will be no benefit.

I will try to change it to `Invalid %q`.

(I almost missed this discussion comment, temporarily reopening)

Reply:
Yea, `\n`'s are needed so that we are not displaying the warning on a used line. And using multiple writes increases binary size. We have 104 bytes free. I don't wan't to play with core anymore. Me ooga booga :b:ython.

Reply:
This was solved.